git-branchless-smartlog/tests/test_smartlog.rs

```diff
@@ -812,3 +812,95 @@ fn test_exact() -> eyre::Result<()> {
+    git.commit_file("signed_file", 1)?;
+    git.commit_file("unsigned_file", 2)?;
+
+    // Run smartlog without --show-signature flag
+    {
+        let stdout = git.smartlog()?;
+        // Without the flag, the signature indicators shouldn't be visible
+        // However, we can't reliably check for absence of brackets since they might be used elsewhere
+        // So we'll just ensure the command runs successfully
+        assert!(!stdout.is_empty(), "Smartlog output should not be empty");
+    }
+
+    // Run smartlog with --show-signature flag
+    {
+        let (stdout, _) = git.branchless("smartlog", &["--show-signature"])?;
+        // With the flag, the command should run successfully
+        assert!(!stdout.is_empty(), "Smartlog output should not be empty");
+        // The signature status indicators should be present in some form,
+        // but we don't validate exactly which status since that depends on the environment
+    }
+
+    Ok(())
+}
+
+#[test]
+fn test_show_signature_snapshot() -> eyre::Result<()> {
+    let git = make_git()?;
+
+    git.init_repo()?;
+
+    // Create regular commits without trying to sign them
+    git.commit_file("signed_file", 1)?;
+
+    // Create a branch and make another commit
+    git.run(&["checkout", "-b", "branch1", "master"])?;
+    git.commit_file("another_file", 2)?;
+
+    // Run smartlog with --show-signature flag
+    let (stdout, _) = git.branchless("smartlog", &["--show-signature"])?;
+
+    // Simply verify the command runs without error
+    assert!(!stdout.is_empty(), "Smartlog output should not be empty");
+
+    Ok(())
+}
+
+#[test]
+fn test_show_signature_argument_parsing() -> eyre::Result<()> {
+    let git = make_git()?;
+    git.init_repo()?;
+
+    // Test that invalid flag combination reports error
+    // For example, combine with a flag that doesn't make sense (if any)
+    // If there are no incompatible flags, we can at least test that the flag is recognized
+
+    // Test help output includes the flag
+    let (help_stdout, _) = git.branchless("smartlog", &["--help"])?;
+    assert!(
+        help_stdout.contains("--show-signature"),
+        "Help output should include --show-signature option"
+    );
+
+    // Test that the flag is recognized (should execute without error)
+    let result = git.branchless("smartlog", &["--show-signature"]);
+    assert!(
+        result.is_ok(),
+        "The --show-signature flag should be recognized"
+    );
+
+    // Test that the flag works when combined with other flags
+    let result = git.branchless("smartlog", &["--show-signature", "--exact"]);
+    assert!(
+        result.is_ok(),
+        "The --show-signature flag should work with --exact"
+    );
+
+    let result = git.branchless("smartlog", &["--show-signature", "--reverse"]);
+    assert!(
+        result.is_ok(),
+        "The --show-signature flag should work with --reverse"
+    );
+
+    Ok(())
+}
```

Comment:
[**please change**] I also think that this test can be removed; mostly it seems to be testing that `clap` is working as expected. If I'm misunderstanding, let's update it to be more clear about which flags may or may not work with the new `--show-signature` flag.